### PR TITLE
Add test coverage for pseudo custom attributes queries via Attribute.*

### LINKED ISSUE #27462: Add test coverage for reflection on PseudoCustomAttributes
Follow up on https://github.com/dotnet/corefx/issues/27426

### PR DESCRIPTION
**Pull request by jkotas**
Fixes #27462

### CODE REVIEW

**Comment by jkotas** [comment]
Blocked on new CoreCLR

File: src/System.Runtime/tests/System/PseudoCustomAttributeTests.cs [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+            yield return new object[] { testMethod, 
+                new DllImportAttribute("nonexistant") { CharSet = CharSet.Unicode, SetLastError = true, EntryPoint = "MyEntryPoint" } };
+
+            yield return new object[] { testMethod, new PreserveSigAttribute() };
+        }
+
+        private static Type TryGetComInterfaceType()
+        {
+            // Ignore the exceptions if COM is not supported
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/PseudoCustomAttributeTests.cs]: Interesting. typeof can fail?  I would have thought a failure here would be at JIT time, in which case the try/catch wouldn't matter, but I guess no?

File: src/System.Runtime/tests/System/PseudoCustomAttributeTests.cs [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+                Assert.True(Attribute.IsDefined(type, attribute.GetType()));
+                Assert.True(Attribute.IsDefined(type, attribute.GetType(), true));
+                break;
+            case MemberInfo memberInfo:
+                Assert.True(Attribute.IsDefined(memberInfo, attribute.GetType()));
+                Assert.True(Attribute.IsDefined(memberInfo, attribute.GetType(), true));
+                break;
+            case ParameterInfo parameterInfo:
+                Assert.True(Attribute.IsDefined(parameterInfo, attribute.GetType()));
+                Assert.True(Attribute.IsDefined(parameterInfo, attribute.GetType(), true));
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/PseudoCustomAttributeTests.cs]: Do we have any tests validating that IsDefined returns false when it's supposed to?

File: src/System.Runtime/tests/System/PseudoCustomAttributeTests.cs (hunk 1 above)
Comment (jkotas) [reply src/System.Runtime/tests/System/PseudoCustomAttributeTests.cs]: I have not actually tested this on Unix and it did not work as you have pointed out.

File: src/System.Runtime/tests/System/PseudoCustomAttributeTests.cs (hunk 2 above)
Comment (jkotas) [reply src/System.Runtime/tests/System/PseudoCustomAttributeTests.cs]: Added.

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by jkotas** [comment]
@dotnet-bot test this please

**Comment by jkotas** [comment]
@dotnet-bot test OSX x64 Debug Build  please

**Pull request merged**